package-res/resources/web/pentaho/visual/base/model.js

```diff
@@ -45,37 +74,239 @@ define([
      * @description Creates a base `Model`.
      * @param {pentaho.visual.base.spec.IModel} modelSpec A plain object containing the model specification.
      */
-    var Model = Complex.extend({
-          meta: {
-            id: "pentaho/visual/base",
-            view: "View",
-            isAbstract: true,
-            props: [
-              {
-                name: "width",
-                type: "number",
-                isRequired: true
-              },
-              {
-                name: "height",
-                type: "number",
-                isRequired: true
-              },
-              {
-                name: "isInteractive",
-                type: "boolean",
-                value: true
-              },
-              {
-                name: "data",
-                type: "object",
-                isRequired: true
-              }
-            ]
+    var Model = Complex.extend(/** @lends pentaho.visual.base.Model# */{
+        //region Event Flows Handling
+        /**
+         * Modifies the current selection filter based on an input filter and on a selection mode.
+         *
+         * This action is the entry point for user-driven modifications of the current selection filter,
```

Comment:
This action is the entry point for user-driven modifications of the current selection filter. For example, if the user clicked a bar in a bar chart or drew a rectangle over a set of bars in a bar chart.